client-app/shared/catalog/composables/useProduct.ts

```diff
@@ -20,6 +24,21 @@ export function useProduct() {
     }
   }
 
+  broadcast.on(productsInWishlistEvent, (eventItems: ProductInWishlistEventDataType[]) => {
+    let trigger = false;
+
+    eventItems.forEach(({ productId, inWishlist }) => {
+      if (product.value && product.value.id === productId) {
+        product.value.inWishlist = inWishlist;
+        trigger = true;
+      }
+    });
+
+    if (trigger) {
+      triggerRef(product);
```

Comment:
triggerRef use with shallowRef.
But product uses ref. It is deeply reactive. 
Should work without triggerRef. 
Please try
https://vuejs.org/api/reactivity-advanced.html#triggerref